skbio/diversity/beta/_fast_unifrac.py

```diff
@@ -0,0 +1,444 @@
+from __future__ import division
+import numpy as np
+from ._unifrac import _validate
+
+
+def unifrac(branch_lengths, i, j):
+    """Calculates unifrac(i,j) from branch lengths and cols i and j of m.
+
+    This is the original, unweighted UniFrac metric.
+
+    Parameters
+    ----------
+    branch_lengths: np.array
+         branch_lengths should be row vector, same length as # nodes in tree.
+    i : np.array
+        a slice of states from m, same length as # nodes in tree.
+        Slicing m (e.g. m[:,i]) returns a vector in the right format; note
+        that it should be a row vector (the default), not a column vector.
+    j : np.array
+        a slice of states from m, same length as # nodes in tree.
+        Slicing m (e.g. m[:,j]) returns a vector in the right format; note
+        that it should be a row vector (the default), not a column vector.
+
+    Returns
+    -------
+    float
+        Unweighted unifrac metric
+
+    Notes
+    -----
+    This is cogent.maths.unifrac.fast_tree.unifrac, but there are
+    other metrics that can (should?) be ported like:
```

Comment:
Since these are less widely used, I think we should get the optimized code in place for the weighted and unweighted first, and then worry about them (we ultimately might not care as much about having optimized and "educational" implementations of these). What do you think @wasade?